libs/sdk-core/src/breez_services.rs

```diff
@@ -1230,33 +1240,39 @@ impl Receiver for PaymentReceiver {
             .ok_or("Failed to retrieve node state")
             .map_err(|err| anyhow!(err))?;
 
+        let amount_sats = req_data.amount_sats;
         let amount_msats = amount_sats * 1000;
 
         let mut short_channel_id = parse_short_channel_id("1x0x0")?;
         let mut destination_invoice_amount_sats = amount_sats;
 
+        let mut channel_opening_fee_params = None;
+        let mut channel_fees_msat = None;
+
         // check if we need to open channel
         let open_channel_needed = node_state.inbound_liquidity_msats < amount_msats;
         if open_channel_needed {
             info!("We need to open a channel");
 
-            // we need to open channel so we are calculating the fees for the LSP
-            let channel_fees_msat_calculated =
-                amount_msats * lsp_info.channel_fee_permyriad as u64 / 10_000 / 1_000_000;
-            let channel_fees_msat = max(
-                channel_fees_msat_calculated,
-                lsp_info.channel_minimum_fee_msat as u64,
-            );
-
-            if amount_msats < channel_fees_msat + 1000 {
-                return Err(anyhow!(
-                    "requestPayment: Amount should be more than the minimum fees {} sats",
-                    lsp_info.channel_minimum_fee_msat / 1000
-                ));
+            // we need to open channel so we are calculating the fees for the LSP (coming either from the user, or from the LSP)
+            let ofp = lsp_info.choose_channel_opening_fees(
+                req_data.opening_fee_params,
+                DynamicFeeType::Cheapest,
+            )?;
```

Comment:
I think this errors in the backward compatible case.
`req_data.opening_fee_params` will be `None` then, and we actually want to send `None` to the LSP then.
So when `None` is passed as an argument, when it comes from redeem_swap, this should not error.

Reply:
This comment is no longer relevant